main.cpp

```diff
@@ -50,8 +50,43 @@ struct Wrapper
     Wrapper(Type&& t) : val(std::move(t)) 
     { 
         std::cout << "Wrapper(" << typeid(val).name() << ")" << std::endl; 
-    }
+    }
+    void print()
+    {
+        std::cout << "Wrapper::print(" << val << ")" << std::endl; 
+    }
+private:
+    Type val;
+};
+
+template <>
+struct Wrapper<Point>
+{
+    Wrapper(Point&& p) : val(std::move(p)) 
+    { 
+        std::cout << "Wrapper(" << typeid(val).name() << ")" << std::endl; 
+    }
+    void print()
+    {
+        std::cout << "Wrapper::print(" << val.toString() << ")" << std::endl; 
+    }
+private:
+    Point val;
 };
+
+template<typename T, typename ... Args>
+void variadicHelper(T&& first, Args&& ... args)
+{
+    Wrapper<T>(std::forward<T>(first)).print();
+    variadicHelper(std::forward<Args>(args)...);
+    
+}
+template<typename T>
+void variadicHelper(T&& first)
+{
+    Wrapper<T>(std::forward<T>(first)).print();
+        
+}
```

Comment:
Project9.1 challenge 2:
There is another recursive solution.  it is possible for `Args ...` to be empty.
your task: remove your implementation for instruction `4)` and implement this alternate solution.
implement it **after** your existing `variadicHelper` function.
hint: what does the **recursive call** actually look like when called with an empty Variadic Parameter list?

Reply:
Updated with new recursion implementation.